hms-app/src/components/admin-tools/UserDetails.tsx

```diff
@@ -0,0 +1,220 @@
+  )
+
+  const editModal = (
+    <Modal
+      centered
+      opened={editModalOpened}
+      onClose={() => setEditModalOpened(false)}
+      withCloseButton={false}
+    >
+      Edit User
+      <UserForm contextID={userID} />
+      {isUserLoading && <div>Loading...</div>}
+      <p>
+        (This window will automatically close as soon as the category is
+        deleted)
+      </p>
+    </Modal>
+  )
+
+  return (
+    <>
+      {isUserError && (
+        <div>
+          <h3>Error loading hackathons</h3>
+          <p>something went wrong.</p>
+        </div>
+      )}
+      {isUserLoading && (
+        <div>
+          <h3>User details are loading...</h3>
+        </div>
+      )}
+
+      {!isUserLoading && !isUserError && (
+        <Card withBorder radius="md" p="md" className={classes.card}>
+          <Card.Section className={classes.section}>
+            <Text mt="md" className={classes.label} color="dimmed">
+              Name
+            </Text>
+            <Text size="md" mt="xs">
+              {user.firstName} {user.lastName}
+            </Text>
+            <Text size={'xs'}>ID: {user.id}</Text>
+            <Text size={'xs'}>
+              Creation Date: {user.creationDate?.toString()}
+            </Text>
+          </Card.Section>
+
+          <Card.Section className={classes.section}>
+            <Text mt="md" className={classes.label} color="dimmed">
+              E-mail
+            </Text>
+            <Text size={'md'}>{user.emailAddress}</Text>
+          </Card.Section>
+
+          <Card.Section className={classes.section}>
+            <Text mt="md" className={classes.label} color="dimmed">
+              Roles
+            </Text>
+            <Group spacing={7} mt={5}>
+              {user.roles?.map((role, index) => (
+                <Badge
+                  color={theme.colorScheme === 'dark' ? 'dark' : 'gray'}
+                  key={index}
+                >
+                  {role}
+                </Badge>
+              ))}
+            </Group>
+          </Card.Section>
+
+          <Card.Section className={classes.section}>
+            <Text mt="md" className={classes.label} color="dimmed">
+              Skills
+            </Text>
+            <Group spacing={7} mt={5}>
+              {user.skills?.map((skill, index) => (
+                <Badge
+                  color={theme.colorScheme === 'dark' ? 'dark' : 'gray'}
+                  key={index}
+                >
+                  {skill.name}
+                </Badge>
+              ))}
+            </Group>
+          </Card.Section>
+
+          <Card.Section className={classes.section}>
+            <Group position="left" mt="xl">
+              {deleteModal}
+              <Button color={'red'} onClick={() => setDeleteModalOpened(true)}>
+                Delete
+              </Button>
+              {editModal}
+              <Button onClick={() => setEditModalOpened(true)}>Edit</Button>
+              {!isUserLoading && (
```

Comment:
this will always evaluate to `true` here because it is within the other `{!isUserLoading` check